src/domain/fiat/index.types.d.ts

```diff
@@ -1,2 +1,47 @@
 type UsdCents = number & { readonly brand: unique symbol }
 type DisplayCurrencyBaseAmount = number & { readonly brand: unique symbol }
+
+type CurrencyBaseAmount = Satoshis | UsdCents
+
+interface DisplayCurrencyConversionRate {
+  fromSats: (amount: Satoshis) => DisplayCurrencyBaseAmount
+  fromCents: (amount: UsdCents) => DisplayCurrencyBaseAmount
+}
+
+interface DealerFns {
+  buyUsdImmediate: (amount: Satoshis) => Promise<UsdCents>
+  buyUsdImmediateFromCents: (amount: UsdCents) => Promise<Satoshis>
+  sellUsdImmediate: (amount: UsdCents) => Promise<Satoshis>
+  sellUsdImmediateFromSats: (amount: Satoshis) => Promise<UsdCents>
+  getBuyUsdQuoteFromCents: (amount: UsdCents) => Promise<Satoshis>
+}
+
+interface AmountFromSatoshis {
+  sats: Satoshis
+}
+
+interface AmountFromCents {
+  cents: UsdCents
+}
+
+type XOR<T1, T2> =
+  | (T1 & { [k in Exclude<keyof T2, keyof T1>]?: never })
+  | (T2 & { [k in Exclude<keyof T1, keyof T2>]?: never })
+
+type CentsXORSats = XOR<AmountFromSatoshis, AmountFromCents>
+
+type OrderType = "immediate" | "quote"
+
+type GetAmountsSendOrReceiveArgs = {
+  walletCurrency: WalletCurrency
+  order: OrderType
+} & CentsXORSats
+
+type GetAmountsSendOrReceiveRet =
+  | NotReachableError
+  | NotImplementedError
+  | {
```

Comment:
Up to know we have typically defined the 'happy path' type first and the error types after.

Also shouldn't it be a `Promise<>`?

Reply:
> looks like so far we define the type and then wrap it in Promise<Ret> in the function signature
indeed what I have done here